### ISSUE TITLE
HANS Dataset: Incorrect `label_list` and `label`.

### ISSUE THREAD

**Issue by HanGuo97** [huggingface/transformers#6179]
## Environment info

     
- `transformers` version:
- Platform: n/a
- Python version: n/a
- PyTorch version (GPU?): n/a
- Tensorflow version (GPU?): n/a
- Using GPU in script?: n/a
- Using distributed or parallel set-up in script?: n/a

### Who can help

@VictorSanh

## Information

Model I am using (Bert, XLNet ...):

The problem arises when using:
* [ ] the official example scripts: (give details below)
* [ ] my own modified scripts: (give details below)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details below)

## To reproduce

1.
2.
3.

### Incorrect `label_list`
See line: https://github.com/huggingface/transformers/blob/master/examples/adversarial/utils_hans.py#L259
```python
    def get_labels(self):
        """See base class."""
        return ["contradiction", "entailment", "neutral"]
```
HANS dataset has only two labels, non-entailment, and entailment, but here three are given. Similarly, when mapping from text label to label-id, the label "non-entailment" (which exists in the task but not in the afore-defined labels), the below line is used. I'm curious if this is intentional? If so, would be great to add a warning/comment as those might cause subtle errors in the future.

https://github.com/huggingface/transformers/blob/master/examples/adversarial/utils_hans.py#L311
```python
label = label_map[example.label] if example.label in label_map else 0
```

### Incorrect `label` index
The below line uses the last column as the label. However, the HANS dataset uses the first column for `label`
https://github.com/huggingface/transformers/blob/master/examples/adversarial/utils_hans.py#L271
```python
            label = line[-1]
```

**Comment by VictorSanh** [comment]
Yes indeed!
Somehow, I was sure it was corrected on the master but I guess it was only on my private branch.
Let me open a PR, thanks for pointing that out @HanGuo97!

**Comment by HanGuo97** [comment]
Great, appreciate the help!

**Comment by prajjwal1** [comment]
@HanGuo97 Could you please confirm if this `label = label[0]` would have any impact on evaluation results ?
